x-pack/plugins/synthetics/server/legacy_uptime/lib/saved_objects/migrations/monitors/8.8.0.ts

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { omit } from 'lodash';
+import { EncryptedSavedObjectsPluginSetup } from '@kbn/encrypted-saved-objects-plugin/server';
+import { SavedObjectUnsanitizedDoc } from '@kbn/core/server';
+import {
+  ConfigKey,
+  SyntheticsMonitorWithSecrets,
+  MonitorFields,
+  BrowserFields,
+  ScheduleUnit,
+} from '../../../../../../common/runtime_types';
+import { ALLOWED_SCHEDULES_IN_MINUTES } from '../../../../../../common/constants/monitor_defaults';
+import {
+  LEGACY_SYNTHETICS_MONITOR_ENCRYPTED_TYPE,
+  SYNTHETICS_MONITOR_ENCRYPTED_TYPE,
+} from '../../synthetics_monitor';
+import { validateMonitor } from '../../../../../routes/monitor_cruds/monitor_validation';
+import {
+  normalizeMonitorSecretAttributes,
+  formatSecrets,
+} from '../../../../../synthetics_service/utils/secrets';
+
+export const migration880 = (encryptedSavedObjects: EncryptedSavedObjectsPluginSetup) => {
+  return encryptedSavedObjects.createMigration<
+    SyntheticsMonitorWithSecrets,
+    SyntheticsMonitorWithSecrets
+  >({
+    isMigrationNeededPredicate: function shouldBeMigrated(
+      doc
+    ): doc is SavedObjectUnsanitizedDoc<SyntheticsMonitorWithSecrets> {
+      return true;
```

Comment:
Shouldn't this be conditional?

Reply:
No. All saved objects need to be migrated.

This is because we're not only changing fields, but changing the saved object definition.

Here on this line you'll see input type and migrated type
https://github.com/elastic/kibana/pull/154010/files#diff-70c4c5687c04bdbf55c2801b90fb3724c1196230e24c5ce21713c037518cae77R74

Migrated type contains the new saved object definition which includes additional attributes to exclude from AAD, including metadata and alerting fields (you actually put the alerting field on the wrong saved object when you included it. You added it to the api key saved object, not the monitor saved object).